cpp/src/arrow/compute/kernels/nth_to_indices_benchmark.cc

```diff
@@ -0,0 +1,62 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "benchmark/benchmark.h"
+
+#include "arrow/compute/kernels/nth_to_indices.h"
+
+#include "arrow/compute/benchmark_util.h"
+#include "arrow/compute/test_util.h"
+#include "arrow/testing/gtest_util.h"
+#include "arrow/testing/random.h"
+
+namespace arrow {
+namespace compute {
+constexpr auto kSeed = 0x0ff1ce;
+
+static void NthToIndicesBenchmark(benchmark::State& state,
+                                  const std::shared_ptr<Array>& values, int64_t n) {
+  FunctionContext ctx;
+  for (auto _ : state) {
+    std::shared_ptr<Array> out;
+    ABORT_NOT_OK(NthToIndices(&ctx, *values, n, &out));
+    benchmark::DoNotOptimize(out);
+  }
+}
```

Comment:
Can you perhaps call `state.SetItemsProcessed` at the end of this function? Also the same for `sort_to_indices_benchmark.cc`

Reply:
Done